builder/frameworks/espidf.py

```
@@ -244,48 +307,322 @@ def extract_flag_name(line):
+        """Generate board-specific sdkconfig settings from board.json manifest."""
+        board_config_flags = []
+
+        # Handle memory type configuration with platformio.ini override support
+        # Priority: platformio.ini > board.json manifest
+        memory_type = None
+        
+        # Check for memory_type override in platformio.ini
+        if hasattr(env, 'GetProjectOption'):
+            try:
+                memory_type = env.GetProjectOption("board_build.memory_type", None)
+            except:
```

Ruff (0.13.3)

Do not use bare `except`

(E722)